### PR TITLE
Create block controller and routes

### PR DESCRIPTION
**Pull request by rowpacmal**
Created the block controller and routes. Summary of what have been done in this branch:
- Created a new file in the controllers folder, named block-controller.mjs
- Created a new file in the routes folder, named block-routes.mjs
- Imported the router to the app.mjs file
- Added getBlockByIndex, getLatestBlock and createBlock functions in block-controlelrs.njs and exported them into block-routes.mjs

### CODE REVIEW

File: backend/controllers/block-controller.mjs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import Block from '../classes/Block.mjs';
+
+// Dummy data for testing...
+const blockchain = { chain: [Block.genesis] };
```
Comment (alcompilor) [inline backend/controllers/block-controller.mjs:4]: You can discard this when merging to production.

You can still "imagine" the presence of a `blockchain` variable that will be imported into the controller from another module, potentially from `app.mjs`.

File: backend/controllers/block-controller.mjs
Comment (alcompilor) [inline backend/controllers/block-controller.mjs:1]: Wait to refactor so it uses the classes we made/planning to make for responses and errors.

**Review by alcompilor — CHANGES_REQUESTED** [review]
Look for comments.
There is a conflict in `app.mjs` as well hope you can pull the latest version of `app.mjs` from `main` branch and fix the conflict in vscode.

**Comment by rowpacmal** [comment]
I removed the test code and comments. I managed to merge the main into the branch and resolve the conflicts with no difficulties.

**Comment by rowpacmal** [comment]
Replaced placeholder responses with ResponseData and ErrorResponse models.

**Review by BTK88 — APPROVED** [review]
# Dont notice any conflict or problem. Good Job!

**Pull request merged**